lua/luasnip/extras/snippet_list.lua

```diff
@@ -0,0 +1,88 @@
+local available = require("luasnip").available
+
+local function snip_info(snippet)
+	return {
+		name = snippet.name,
+		trigger = snippet.trigger,
+		description = snippet.dscr,
+		wordTrig = snippet.wordTrig and true or false,
+		regTrig = snippet.regTrig and true or false,
+		docstring = snippet:get_docstring(),
+	}
+end
+
+local function get_name()
+	return "LuaSnip://Snippets"
+end
+
+local win_opts = { foldmethod = "indent" }
+local buf_opts = { filetype = "lua" }
+
+local function set_win_opts(win, opts)
+	for opt, val in pairs(opts) do
+		vim.api.nvim_win_set_option(win, opt, val)
+	end
+end
+
+local function set_buf_opts(buf, opts)
+	for opt, val in pairs(opts) do
+		vim.api.nvim_buf_set_option(buf, opt, val)
+	end
+end
+
+local function make_scratch_buf(buf)
+	local opts = {
+		buftype = "nofile",
+		bufhidden = "wipe",
+		buflisted = false,
+		swapfile = false,
+		modified = false,
+		modeline = false,
+	}
+
+	set_buf_opts(buf, opts)
+end
+
+local function default_open(printer_result)
```

Comment:
I'd like to make it easier to adjust the "common" settings, like name of the buffer, or options (basically what we had earlier)
I think the best way to achieve this is to create a function which takes the options `win/buf_opts,get_name`, and returns `default_open` as a closure.
Basically
```lua
local function default_open(opts)
	-- maybe extend our options with opts.win/buf_opts instead? (overwrite on
	-- collision)
	local win_opts = opts.win_opts or win_opts
	local buf_opts = opts.buf_opts or buf_opts
	local get_name = opts.get_name or get_name

	return function(printer_result)
		-- use win_opts, buf_opts, get_name in here.
	end
```

the default (in `open_snippet_list`) could then just be `default_open()`.

Reply:
Oh I see! But wait wouldn't this be redundant since no one is calling default_open besides us no? Unless we pass ```opts``` to ```default_open``` too.

Cause otherwise it would look like this

```lua
local function open_snippet_list(opts)
	opts = opts or {}
	opts.snip_info = opts.snip_info or snip_info
	opts.printer = opts.printer or vim.inspect
	opts.open = opts.open or default_open()

	-- load snippets before changing windows/buffers
	local snippets = available(opts.snip_info)

	-- open snippets
	opts.open(opts.printer(snippets))
end
```

So options are never passed to ```default_open``` atm.

Reply:
Ah, true, right now it wouldn't make a difference
We'd have to expose `default_open` :D
Then one can pass other defaults to it, and pass the generated function to `open_snippet_list`

Reply:
Exactly! Check https://github.com/L3MON4D3/LuaSnip/pull/652/commits/cbad3d030396415dc5e2d1d1fb79604937a02990

Now with this we can do something like

```lua
local snippet_list = require('luasnip.extras.snippet_list')
local open_snippet_list = snippet_list.open_snippet_list
local default_open = snippet_list.default_open

    open_snippet_list({
        open = default_open({
            buf_opts = {
                filetype = 'javascript',
            },
        }),
    })
```
As an example. Pretty neat. Of course the user can also define their own ```open``` function as well!

Reply:
Niiice, that's great :ok_hand: